src/lyrics/lyricsproviders.cpp

```diff
@@ -133,7 +133,7 @@ void LyricsProviders::RemoveProvider(LyricsProvider *provider) {
 
 void LyricsProviders::ProviderDestroyed() {
 
-  LyricsProvider *provider = static_cast<LyricsProvider*>(sender());
+  LyricsProvider *provider = dynamic_cast<LyricsProvider*>(sender());
```

Comment:
`dynamic_cast` does not work here when `sender()` is destroyed, that's the reason `qobject_cast` wasn't used here in the first place. This will crash.